crates/iceberg/src/spec/values.rs

```diff
@@ -0,0 +1,417 @@
+                PrimitiveType::Fixed(len) => Ok(Value::Fixed(*len as usize, Vec::from(bytes))),
+                PrimitiveType::Binary => Ok(Value::Binary(Vec::from(bytes))),
+                _ => Err(Error::new(
+                    crate::ErrorKind::ValueByteConversionFailed,
+                    "Converting bytes to decimal is not supported.",
+                )),
+            },
+            _ => Err(Error::new(
+                crate::ErrorKind::ValueByteConversionFailed,
+                "Converting bytes to non-primitive types is not supported.",
+            )),
+        }
+    }
+
+    /// Get datatype of value
+    pub fn datatype(&self) -> Type {
+        match self {
+            Value::Boolean(_) => Type::Primitive(PrimitiveType::Boolean),
+            Value::Int(_) => Type::Primitive(PrimitiveType::Int),
+            Value::LongInt(_) => Type::Primitive(PrimitiveType::Long),
+            Value::Float(_) => Type::Primitive(PrimitiveType::Float),
+            Value::Double(_) => Type::Primitive(PrimitiveType::Double),
+            Value::Date(_) => Type::Primitive(PrimitiveType::Date),
+            Value::Time(_) => Type::Primitive(PrimitiveType::Time),
+            Value::Timestamp(_) => Type::Primitive(PrimitiveType::Timestamp),
+            Value::TimestampTZ(_) => Type::Primitive(PrimitiveType::Timestamptz),
+            Value::Fixed(len, _) => Type::Primitive(PrimitiveType::Fixed(*len as u64)),
+            Value::Binary(_) => Type::Primitive(PrimitiveType::Binary),
+            Value::String(_) => Type::Primitive(PrimitiveType::String),
+            Value::UUID(_) => Type::Primitive(PrimitiveType::Uuid),
+            Value::Decimal(dec) => Type::Primitive(PrimitiveType::Decimal {
+                precision: 38,
+                scale: dec.scale(),
+            }),
+            _ => unimplemented!(),
+        }
+    }
+
+    /// Convert Value to the any type
+    pub fn into_any(self) -> Box<dyn Any> {
+        match self {
+            Value::Boolean(any) => Box::new(any),
+            Value::Int(any) => Box::new(any),
+            Value::LongInt(any) => Box::new(any),
+            Value::Float(any) => Box::new(any),
+            Value::Double(any) => Box::new(any),
+            Value::Date(any) => Box::new(any),
+            Value::Time(any) => Box::new(any),
+            Value::Timestamp(any) => Box::new(any),
+            Value::TimestampTZ(any) => Box::new(any),
+            Value::Fixed(_, any) => Box::new(any),
+            Value::Binary(any) => Box::new(any),
+            Value::String(any) => Box::new(any),
+            Value::UUID(any) => Box::new(any),
+            Value::Decimal(any) => Box::new(any),
+            _ => unimplemented!(),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+
+    use super::*;
+
+    #[test]
+    pub fn boolean() {
+        let input = Value::Boolean(true);
+
+        let raw_schema = r#"{"type": "boolean"}"#;
```

Comment:
Unfortunately, the primitive types are strings and not a dictionary: https://iceberg.apache.org/spec/#schemas
```suggestion
        let raw_schema = r#""boolean""#;
```